pkg/compactor/retention/retention_test.go

```diff
@@ -825,6 +852,9 @@ func TestMarkForDelete_SeriesCleanup(t *testing.T) {
 			expectedModified: []bool{
 				true, true,
 			},
+			numChunksDeleted: []int64{
+				1, 0,
+			},
```

Comment:
should we add another test here where the chunk is partially deleted from two tables, but only gets marked in the oldest one?

Reply:
had a chat with sandeep offline, this test already covers the above scenario. Without the chunk start time check, we would have marked the chunk from both tables after a rewrite